src/States/Elytra.cs

```diff
@@ -466,9 +468,11 @@ private static void Mod_Player_UpdateSprite(On.Celeste.Player.orig_UpdateSprite
                 frame -= (int)(t * (FRAME_COUNT - STABLE_FRAME - 1));
             else
                 frame -= (int)(t * STABLE_FRAME);
-        }
-
-        self.Sprite.SetAnimationFrame(Calc.Clamp(frame, 0, FRAME_COUNT - 1));
+        }
+        if (self.Sprite.CurrentAnimationTotalFrames !< frame)

```

Comment:
I'll be real, I have no idea what does the `!<` do. is it the same as `>=` or just `<`. And is there a way to represent it in a more conventional.

Reply:
Why is this condition mandatory? Doesn't `Clamp` handle the case where `frame` is larger than `CurrentAnimationTotalFrames`?

Reply:
I can't figure out how to change the comment lines :sweat_smile: I am talking about line 472.

Reply:
> Is this condition mandatory? Doesn't `Clamp` handle the case where `frame` is larger than `CurrentAnimationTotalFrames`?

oh that is want to prevent animation from not existing in those frames by skin.

Reply:
Ah I see, shouldn't the condition be flipped then? because right now it will only call `SetAnimationFrame` if `frame` is larger then `CurrentAnimationTotalFrames`, or just  change the clamp values to `0` and `self.Sprite.CurrentAnimationTotalFrames - 1`.

Reply:
> shouldn't the condition be flipped then? because right now it will only call `SetAnimationFrame` if `frame` is larger then `CurrentAnimationTotalFrames`

i guess "self.Sprite.CurrentAnimationTotalFrames !< frame" be like "!(self.Sprite.CurrentAnimationTotalFrames < frame)"... so what?

Reply:
and sorry for replied now, I'm relatively busy